wrappers/python/simtk/openmm/app/charmmparameterset.py

```diff
@@ -175,7 +175,33 @@ def loadSet(cls, tfile=None, pfile=None, sfiles=[]):
                 inst.readStreamFile(sfile)
         return inst
 
-    def readParameterFile(self, pfile):
+    def _read_atom_types_from_psf( self, psf ):
+       if( isinstance( psf, str ) ):
+           data = CharmmPsfFile( psf )
+       else:
+           data = psf
+       idx=1000
```

Comment:
Are you sure this doesn't collide with other atom type indices?  My particular worry is from CHARMM CGenFF.  Although I've been told CHARMM is basically doing away with atom type indices and moving to the XPLOR-style PSF files (so indices won't matter much).

Perhaps `idx` can be assigned from `max(self.atom_types_int.keys())` (and adding 1000 to that, maybe), and just make sure that the PSF is the last file parsed when given in a list of files?